src/components/Footer.tsx

```diff
@@ -0,0 +1,85 @@
+import classNames from 'classnames';
+import React from 'react';
+import { deleteTodo } from '../api/todos';
+import { FilterType } from '../types/FilterTypes';
+import { Todo } from '../types/Todo';
+
+export type Props = {
+  todos: Todo[],
+  setSortType: (arg: FilterType) => void,
+  sortType: FilterType,
+  removeTodos: (todoIds: number[]) => void
+};
+
+export const Footer: React.FC<Props> = ({
+  todos, setSortType, sortType, removeTodos,
+}) => {
+  const clearCompletedTodos = async () => {
+    const todosToDelete = todos.filter(todo => todo.completed);
+
+    await Promise.all(
+      todosToDelete.map(
+        todo => deleteTodo(todo.id),
+      ),
+    );
+
+    removeTodos(todosToDelete.map(todo => todo.id));
+  };
```

Comment:
Why do we need to pass an array with all todos ?
Let's create function clearCompletedTodos and pass it to the Footer